### PR TITLE
Updating OpenFlow default port to 6653

### LINKED ISSUE #545: Update Openflow port to 6653
We should move from 6633 to 6653 to match IANA assigned port number.

http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=openflow

### PR DESCRIPTION
**Pull request by bocon13**
- Pass 6653 to controllers that Mininet starts
- Try to connect first on 6653 for RemoteController, then fallback to 6633

fixes #545

### CODE REVIEW

File: mininet/node.py [hunk 1]
```diff
@@ -1498,12 +1498,30 @@ def stop( self ):
...
+        if self.port is None:
+            self.port = 6653
+            warn( "Setting remote controller"
+                  " to %s:%d\n" % ( self.ip, self.port ))
+
+    def isListening( self, ip, port ):
+        "Check if a remote controller is listening at a specific ip and port"
+        listening = self.cmd( "echo A | telnet -e A %s %d" % ( ip, port ) )
         if 'Connected' not in listening:
             warn( "Unable to contact the remote controller"
```
Comment (bocon13) [inline mininet/node.py]: We might reconsider the placement of the log messages.

File: mininet/node.py [hunk 2]
```diff
@@ -1498,12 +1498,30 @@ def stop( self ):
...
+                if self.isListening( self.ip, port ):
+                    self.port = port
+                    info( "Connecting to remote controller"
+                          " at %s:%d\n" % ( self.ip, self.port ))
+                    break
+
+        if self.port is None:
+            self.port = 6653
+            warn( "Setting remote controller"
+                  " to %s:%d\n" % ( self.ip, self.port ))
```
Comment (bocon13) [inline mininet/node.py]: Note: This will break people that start up Mininet before their remote controller, which is listening on 6633.

**Comment by lantz** [comment]
Yeah that seems like an annoying problem. I wonder if we should change the reference controller to listen on the new port, or??

File: mininet/node.py [hunk 3]
```diff
@@ -1498,12 +1498,30 @@ def stop( self ):
 
     def checkListening( self ):
         "Warn if remote controller is not accessible"
-        listening = self.cmd( "echo A | telnet -e A %s %d" %
-                              ( self.ip, self.port ) )
+        if self.port is not None:
+            self.isListening( self.ip, self.port )
+        else:
+            for port in [ 6653, 6633 ]:
```
Comment (lantz) [inline mininet/node.py]: braces not needed here

**Pull request merged**